crates/jstzd/src/task/octez_rollup.rs

```diff
@@ -0,0 +1,114 @@
+use std::{
+    fs::{self, create_dir_all},
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+
+use crate::task::child_wrapper::ChildWrapper;
+
+use super::{child_wrapper::SharedChildWrapper, Task};
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use octez::r#async::{
+    directory::Directory,
+    rollup::{OctezRollupConfig, RollupDataDir},
+};
+use serde::Deserialize;
+
+#[derive(Clone)]
+pub struct OctezRollup {
+    inner: SharedChildWrapper,
+    config: OctezRollupConfig,
+    // holds the TempDir instance so that the directory does not get deleted too soon
+    _data_dir: Arc<Directory>,
+}
+
+#[derive(Debug, Deserialize)]
+struct HealthCheckResponse {
+    healthy: bool,
+}
+
+#[async_trait]
+impl Task for OctezRollup {
+    type Config = OctezRollupConfig;
+
+    async fn spawn(config: Self::Config) -> Result<Self> {
+        let data_dir = match &config.data_dir {
+            RollupDataDir::Temp {
+                preimages_dir: from,
+            } => {
+                let temp_dir = Directory::default();
+                let to = PathBuf::from(&temp_dir).join("wasm_2_0_0");
```

Comment:
`wasm_2_0_0` should not be hardcoded. This depends on the type of the rollup to be executed by the node.